disnake/iterators.py

```diff
@@ -763,7 +763,7 @@ async def _retrieve_guilds_after_strategy(self, retrieve):
 
 
 class MemberIterator(_AsyncIterator["Member"]):
-    def __init__(self, guild, limit=1000, after=None) -> None:
+    def __init__(self, guild, limit: int = 1000, after=None) -> None:
```

Comment:
```suggestion
    def __init__(self, guild, limit: Optional[int] = 1000, after=None) -> None:
```
the type annotation of `Guild.fetch_members` is also incorrect.